extensions/cache/runtime/src/main/java/io/quarkus/cache/build/CacheBuildRecorder.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.cache.build;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.cache.augmented.interceptor.CacheRepository;
+import io.quarkus.cache.augmented.interceptor.CaffeineCache;
+import io.quarkus.runtime.annotations.Recorder;
+
+/**
+ * @author Gwenneg Lepage
+ */
+@Recorder
+public class CacheBuildRecorder {
+
+    private static final Logger LOGGER = Logger.getLogger(CacheBuildRecorder.class);
+
+    public void buildCaches(BeanContainer beanContainer, Set<CacheInfo> cacheInfos) {
+        Map<String, CaffeineCache> caches = new HashMap<>(cacheInfos.size() + 1, 1.0F);
+
+        cacheInfos.forEach(cacheInfo -> {
```

Comment:
We try to avoid lambdas in runtime code. Especially when it's not useful at all (a for loop would be more readable).

Lambdas have a non negligible cost in memory consumption.